BidMyAdz Bid Adapter: add new bid adapter

bidmyadz opened the pull request:
## Type of change

- [x]  New bidder adapter 

## Description of change
<!-- Describe the change proposed in this pull request -->
Adding BidMyadz prebid adapter

<!-- For new bidder adapters, please provide the following -->
- test parameters for validating bids
```
adUnits = [
                {
                    code: 'adunit1',
                    mediaTypes: {
                        banner: {
                            sizes: [ [300, 250], [320, 50] ],
                        }
                    },
                    bids: [
                        {
                            bidder: 'bidmyadz',
                            params: {
                                placementId: 'test',
                                traffic: 'banner'
                            }
                        }
                    ]
                },
                {
                    code: 'addunit2',
                    mediaTypes: {
                        video: {
                            playerSize: [640, 480],
                            context: 'instream'
                        }
                    },
                    bids: [
                        {
                            bidder: 'bidmyadz',
                            params: {
                                placementId: 'test',
                                traffic: 'video'
                            }
                        }
                    ]
                },
                {
                    code: 'addunit3',
                    mediaTypes: {
                        native: {
                            title: {
                                required: true
                            },
                            body: {
                                required: true
                            },
                            icon: {
                                required: true,
                                size: [64, 64]
                            }
                        }
                    },
                    bids: [
                        {
                            bidder: 'bidmyadz',
                            params: {
                                placementId: 'test',
                                traffic: 'native'
                            }
                        }
                    ]
                }
            ];
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer: contact@bidmyadz.com
- [x] official adapter submission
- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/pull/3041

ChrisHuie commented:
@bidmyadz can you please add support for advertiserdomain meta tag in this new adapter pr 🙏 Starting in the last release we now require adomain support

aleksatr commented:
@bidmyadz instead of using 'traffic' param, why don't you check/read what media types are defined on the given ad unit (or the bid request in your case), `traffic` param seems redundant to me?

aleksatr commented on modules/bidmyadzBidAdapter.js:
```diff
@@ -0,0 +1,151 @@
+const AD_URL = 'https://endpoint.bidmyadz.com/pbjs';
+const SYNC_URL = 'https://cookie-sync.bidmyadz.com';
+
+function isBidResponseValid (bid) {
```
isBidResponseValid() function should catch this:
![image](https://user-images.githubusercontent.com/12955750/122932133-c4b4a080-d36d-11eb-9aed-cbc1b980b985.png)

bidmyadz replied:
Added additional check.

bidmyadz commented:
> @bidmyadz can you please add support for advertiserdomain meta tag in this new adapter pr 🙏 Starting in the last release we now require adomain support

I don't quite understand what it means to support adomain.
If our bidder always returns meta.advertiserDomains. We added a check for the return of the meta object to the tests. We also added a check for the meta object to the response validation. Does this mean that we support him or something else needs to be added ?

bidmyadz commented:
> @bidmyadz instead of using 'traffic' param, why don't you check/read what media types are defined on the given ad unit (or the bid request in your case), `traffic` param seems redundant to me?
> Especially that this way you have a way of supporting more than one media type per ad unit (you can bid on video, banner & native for the one single ad unit).

@aleksatr 
Our system requires that one placement has only one mediaType. If we remove traffic, then our publishers will not be able to install our adapter in multi-format ad units. (The adapter will not know what mediaType is expected from it.) What could be a problem for us.

bidmyadz commented:
> @bidmyadz instead of using 'traffic' param, why don't you check/read what media types are defined on the given ad unit (or the bid request in your case), `traffic` param seems redundant to me?
> Especially that this way you have a way of supporting more than one media type per ad unit (you can bid on video, banner & native for the one single ad unit).
@aleksatr 

In our system, placements can have only one adFormat.

ChrisHuie commented:
@bidmyadz Sorry. Here is the specific issue for adomain in Prebid 5.0 #6650

ChrisHuie commented:
@bidmyadz following up on this pr for the 'utils' imports. Looks like everything else is fine though. If you could pull in recent commits too that would be appreciated :)

ChrisHuie commented:
@bidmyadz closing this pr for now. Please feel free to reopen when it is again ready for review :)
